### PR TITLE
implemented hardware pid and facilitated tuning

### PR DESCRIPTION
**Pull request by matt-mekha**
also moved nominal/peak outputs (fancy motor speak for min/max) to hardware

### CODE REVIEW

File: botplus/build.gradle [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies {
 }
 /// ------ End No touchy ------ ///
```
Comment (DGKSK8LIFE) [inline botplus/build.gradle:31]: Yeah please remove this comment

**Review by DGKSK8LIFE — APPROVED** [review]

File: botplus/build.gradle (hunk 1 above)
Comment (matt-mekha) [reply botplus/build.gradle:31]: @AlexDvorak

File: botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt [hunk 2]
```diff
@@ -6,27 +6,74 @@ import com.ctre.phoenix.motorcontrol.can.TalonFX
...
+  init {
+    motorController.configFactoryDefault()
+    motorController.configSelectedFeedbackSensor(specs.feedbackDevice)
+  }
+
+  /**
+   * Returns the maximum revolutions per second of the output shaft. The real-world value is
+   * probably lower because this method uses the motor's free speed.
+   */
+  fun getMaxRevsPerSecond() = specs.freeSpeed * gearRatio
```
Comment (AlexDvorak) [inline botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt]: `specs.freeSpeed` is stored as RPM, not RPS

File: botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt [hunk 3]
```diff
@@ -35,61 +82,66 @@ data class MotorSpecs(
     val stallTorque: Double, // newton-meters
     val stallCurrent: Double, // amps
     val feedbackDevice: FeedbackDevice,
-    val encoderUnitRatio: Double
+    val encoderUnits: Double
```
Comment (AlexDvorak) [inline botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt:88]: could this be `encoderUnitsPerRevolution`?

**Review by AlexDvorak — CHANGES_REQUESTED** [review]

File: botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt (hunk 3 above)
Comment (matt-mekha) [reply botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt:88]: too verbose in my opinion; the library users will never have to see this anyway

File: botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt (hunk 2 above)
Comment (matt-mekha) [reply botplus/src/main/kotlin/org/awtybots/frc/botplus/motors/Motors.kt]: good catch

**Review by AlexDvorak — APPROVED** [review]

**Pull request merged**